src/contracts/HypMinter.sol

```diff
@@ -102,41 +115,56 @@ contract HypMinter is AccessManagedUpgradeable {
-     * @dev Mints the full MINT_AMOUNT and splits it between stakers and operators based on operatorBps
      */
-    function mintAndDistribute() external {
+    function mint() external {
         require(block.timestamp >= mintAllowedTimestamp, "HypMinter: Minting not started");
 
+        // Calculate next epoch timestamp (30 days after last mint)
+        uint256 newTimestamp = lastRewardTimestamp + 30 days;
+        require(block.timestamp >= newTimestamp, "HypMinter: Epoch not ready");
+
+        // Update the last mint timestamp for next epoch calculation
+        rewardDistributions[newTimestamp].minted = true;
+        lastRewardTimestamp = newTimestamp;
+
         // Mint the full amount to this contract
         HYPER.mint(address(this), MINT_AMOUNT);
         emit Mint();
-
-        // Distribute rewards to both stakers and operators
-        _distributeStakingRewards();
-        _distributeOperatorRewards();
-        emit Distribution(operatorBps);
     }
 
-    /**
-     * @notice Distributes the staking portion of minted tokens to the rewards contract
-     * @dev Enforces 30-day epochs and updates the lastMintTimestamp
-     * @dev Distributes tokens to the REWARDS contract for the SYMBIOTIC_NETWORK
-     */
-    function _distributeStakingRewards() internal {
-        // Calculate next epoch timestamp (30 days after last mint)
-        uint256 newTimestamp = lastMintTimestamp + 30 days;
-        require(block.timestamp >= newTimestamp, "HypMinter: Epoch not ready");
+    function distributeRewards(
+        uint256 rewardTimestamp
+    ) external {
+        // Check if the distribution is ready
+        require(block.timestamp >= rewardTimestamp + distributionDelay, "HypMinter: Distribution not ready");
 
-        // Update the last mint timestamp for next epoch calculation
-        lastMintTimestamp = newTimestamp;
+        // Check if timestamp is valid and not already distributed
+        DistributionInfo memory distributionInfo = rewardDistributions[rewardTimestamp];
+        require(distributionInfo.minted, "HypMinter: Rewards not minted");
+        require(!distributionInfo.distributed, "HypMinter: Rewards already distributed");
+
+        // Update the distribution info
+        rewardDistributions[rewardTimestamp].distributed = true;
 
-        // Distribute only the staking portion of minted tokens to the rewards contract
+        // Distribute staking rewards
         REWARDS.distributeRewards({
             network: SYMBIOTIC_NETWORK,
             token: address(HYPER),
             amount: getStakingMintAmount(),
-            data: abi.encode(newTimestamp, type(uint256).max, bytes(""), bytes(""))
+            data: abi.encode(rewardTimestamp, type(uint256).max, bytes(""), bytes(""))
         });
+
+        // Distribute operator rewards
+        HYPER.transfer(operatorRewardsManager, getOperatorMintAmount());
+        emit Distribution(operatorBps);
```

Comment:
we want the funds to be minted to the operator rewards manager before we distribute staking rewards, otherwise we will have no time to fund the merkle distributors